### ISSUE TITLE
KerasClassifier model produces poor performance

### ISSUE THREAD

**Issue by drodriguez3** [Trusted-AI/adversarial-robustness-toolbox#1176]
**Describe the bug**
I have a pretrained keras model and I'm using the KerasClassifier function as a wrapper, the resulting model produces poor performance. The model achieves 97% accuracy before using the wrapper but after using the wrapper the performance of the model reduces to 52% accuracy. 

**To Reproduce**
import keras
from keras.models import load_model
from art.estimators.classification import KerasClassifier
from sklearn.metrics import accuracy_score

model = load_model(pathModel)
model = KerasClassifier(model)
y_pred = model.predict(x_test, x_test.shape[0], False)
y_pred = np.argmax(y_pred, axis=1)
y_pred = keras.utils.to_categorical(y_pred, num_classes = 2)
accuracy = accuracy_score(y_test2, y_pred2)
print(accuracy)
**Expected behavior**
I expect the wrapped model to produce similar performance to the unwrapped version of the same model.

**System information (please complete the following information):**
python 3.7.6
Keras 2.3.1
adversarial-robustness-toolbox 1.7.0

**Comment by beat-buesser** [comment]
Hi @drodriguez3 I think this might be related to extensions for binary classification in ART 1.7.0. Could you please describe you label format and model in more detail?

**Comment by drodriguez3** [comment]
My model is a Keras resent 50 classifier with soft max trained on chest X-ray images to classify pneumonia and non-pneumonia images. The original model was not trained using ART toolbox.

Thank you
David Rodriguez

**Comment by beat-buesser** [comment]
Hi @drodriguez3 Thank you, could you please provide more details like the shape of the labels and the output shape of your model, etc.?

**Comment by drodriguez3** [comment]
The dataset used for training can found at https://www.kaggle.com/paultimothymooney/chest-xray-pneumonia and the standard pretrained keras model is attached. The model summary input is input_1 (InputLayer) (None, 224, 224, 3) and the output is dense_1 (Dense) (None, 2).
The one hot encoded label shapes for train, validate and test sets are listed below, respectively. (2253, 2)
(460, 2)(461, 2)

Thank youDavid Rodriguez
 

    On Monday, June 21, 2021, 10:27:01 AM CDT, Beat Buesser ***@***.***> wrote:  
 
 


Hi @drodriguez3 Thank you, could you please provide more details like the shape of the labels and the output shape of your model, etc.?

**Comment by beat-buesser** [comment]
Ok, the shapes look good. Could you please share the code in a Google Colab? That way we could try it out to see what is happening.
